libraries/AP_Airspeed/AP_Airspeed_ND210.cpp

```diff
@@ -0,0 +1,302 @@
+ * When enabled, the I/O watchdog will monitor the I/O activity.
+ * If I/O activity is not detected for the I/O Watchdog timeout time, the pressure sensor will reset itself.
+ * The I/O watchdog timeout time is determined by the currently active bandwidth setting.
+ *
+ * Bandwidth Limit Select, 3 bits
+ * Bits 4-6 control the BW Limit Filter.
+ * 000 1 Hz
+ * 001 2 Hz
+ * 010 5 Hz
+ * 011 10 Hz
+ * 100 20 Hz
+ * 101 50 Hz
+ * 110 100 Hz
+ * 111 200 Hz
+ *
+ * Bit 7 is the Notch Filter Enable bit. When enabled, the 50/60Hz notch filter is active *
+ *
+ *  ModeControlRegister default value is 0xF6 (b1 111 0 110 ) Notch (1), 200Hz (b111), No Watchdog, PressureRange = 5 (b110)
+ *  (NotchFilterEnable << 7) | (BandWidth << 6) | (WatchdogEnable << 4) | PressureRange;
+ */
+static constexpr uint8_t MODE_CONTROL_REGISTER = 0xF7; // 0xF6 is default
+
+/*
+ * The Rate Control Register controls the rate at which the DAV
+ * pin is asserted indicating new data is available. This register
+ * is primarily used to throttle down the actual data transfer
+ * rate (when using the DAV as the trigger to sample) since the
+ * general industrial requirement is less than the internal 444Hz update rate.
+ * The value of the Rate Control Register is
+ * the divisor of the 444 Hz internal data rate. Since a divisor of
+ * zero is not possible, a zero value will select the auto-select
+ * rate mode. In this mode, the rate is selected based on the
+ * selected bandwidth limit. The auto rate value is roughly 3x
+ * the corner frequency of the selected bandwidth limit in all
+ * auto selected rates (where possible).
+ * 00000000 = AutoSelect (based on BW)
+ * 00000001 = 444Hz
+ * 00000010 = 222Hz ...
+ */
+static constexpr uint8_t RATE_CONTROL_REGISTER = 0x00; // Auto Select aka (BW = 1Hz, so Rate will be 3Hz)
+
+extern const AP_HAL::HAL &hal;
+
+AP_Airspeed_ND210::AP_Airspeed_ND210(AP_Airspeed &_frontend, uint8_t _instance) :
+    AP_Airspeed_Backend(_frontend, _instance)
+{
+}
+
+/*
+  send a 16 bit command code
+ */
+bool AP_Airspeed_ND210::_send_configuration_command()
+{
+    uint8_t b[2] {MODE_CONTROL_REGISTER, RATE_CONTROL_REGISTER};
+    return _dev->transfer(b, 2, nullptr, 0);
+}
+
+// probe and initialise the sensor
+bool AP_Airspeed_ND210::init()
+{
+    _dev = hal.i2c_mgr->get_device(get_bus(), ND210_I2C_ADDR);
+    if (!_dev) {
+        GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "ND210 no sensor found at 0x%02X", ND210_I2C_ADDR);
+        return false;
+    }
+    _dev->get_semaphore()->take_blocking();
+    // lots of retries during probe
+    _dev->set_retries(10);
+
+    // set Mode and Rate modes
+    if (!_send_configuration_command()) {
+        _dev->get_semaphore()->give();
+        GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "ND210[%u]: Init Send Command Failed.", get_instance());
+        return false;
+    }
+
+    _dev->get_semaphore()->give();
+
+    switch(MODE_CONTROL_REGISTER & 0x07) {
```

Comment:
Could we get names for these constants, please?  Defines or const vairables in the cpp file...